velox/common/memory/Memory.h

```diff
@@ -630,11 +638,11 @@ void* FOLLY_NULLABLE MemoryPoolImpl<Allocator, ALIGNMENT>::reallocate(
       ALIGNER<ALIGNMENT>{}, p, alignedSize, alignedNewSize);
   if (UNLIKELY(!newP)) {
     free(p, alignedSize);
-    auto errorMessage = fmt::format(
-        MEM_CAP_EXCEEDED_ERROR_FORMAT,
-        succinctBytes(cap_),
-        succinctBytes(difference));
-    VELOX_MEM_CAP_EXCEEDED(errorMessage);
+    VELOX_FAIL(
```

Comment:
Why are you changing VELOX_MEM_CAP_EXCEEDED to VELOX_FAIL? Don't we need to throw an exception with the "right" error code?